### PR TITLE
cache: add a function for FIFO

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #5257 

### What is changed and how does it work?
add a function to get continuous items that have the same comparable attribute with the the lastest one.

```commit-message
get continuous items that have the same comparable attribute with the the lastest one.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/cache/cache_test.go [hunk 1]
```diff
@@ -221,6 +221,41 @@ func TestFifoCache(t *testing.T) {
 	re.Equal(0, cache.Len())
 }
 
+type testStruct struct {
```
Comment (rleungx) [inline pkg/cache/cache_test.go:224]: How about putting it inside of the test function?

File: pkg/cache/fifo.go [hunk 2]
```diff
@@ -89,6 +89,25 @@ func (c *FIFO) FromElems(key uint64) []*Item {
 	return elems
 }
 
+// FromLastSameElems returns continuous items that have the same comparable attribute with the the lastest one.
+func (c *FIFO) FromLastSameElems(checkFunc func(interface{}) bool, comparableFunc func(interface{}) string) []*Item {
```
Comment (bufferflies) [inline pkg/cache/fifo.go:93]: It is not friendly to use two function to check item. howa bout  checkFunc func(interface{})bool,string

**Review by rleungx — APPROVED** [review]
LGTM, and maybe we can make it more general in the future.

**Review by bufferflies — APPROVED** [review]

**Pull request merged**